### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by Eliacus**

### CODE REVIEW

**Comment by Eliacus** [comment]
#2 #3

File: src/DCGAN.py [hunk 1]
```diff
@@ -6,61 +6,72 @@
...
+                num_generator_features * 2, num_generator_features, 4, 2, 1, bias=False
+            ),
+            nn.BatchNorm2d(num_generator_features),
             nn.ReLU(True),
-            # state size. (ngf) x 32 x 32
-            nn.ConvTranspose2d(ngf, nc, 4, 2, 1, bias=False),
+            # state size. (num_generator_features) x 32 x 32
+            nn.ConvTranspose2d(num_generator_features, num_channels, 4, 2, 1, bias=False),
             nn.Tanh()
-            # state size. (nc) x 64 x 64
```
Comment (Samicus) [inline src/DCGAN.py:36]: Is the output dimension always num_channels x 64 x 64? If we want to try our model on a dataset with different resolution, can we make this process easier by adding desired output shape as an input to the discriminator or will we need to create a new model for it?

**Review by Samicus — COMMENTED** [review]
One comment on the overall structure, otherwise I like the new naming convensions and the refactorizations

File: src/DCGAN.py (hunk 1 above)
Comment (Eliacus) [reply src/DCGAN.py:36]: We need to create new models for it *I think*. The number of deconvolutions needed to project the image to 1d changes, so the depth of the network changes. In either case, we have #9 to implement that for us

Comment (Samicus) [reply src/DCGAN.py:36]: Alright, sounds good

**Review by Samicus — APPROVED** [review]

**Pull request merged**